src/managers/run-strategies/reuse-last-initialized.js

```diff
@@ -69,13 +69,14 @@ export default class ReuseLastInitializedStrategy {
         const sessionFilter = injectFiltersFromSession(this.options.flag, userSession, this.options.scope);
         const runopts = runService.getCurrentConfig();
         const filter = $.extend(true, { trashed: false }, sessionFilter, { model: runopts.model });
-        return runService.query(filter, { 
+        return runService.query(filter, {
             startrecord: 0,
             endrecord: 0,
-            sort: 'created', 
+            sort: 'created',
             direction: 'desc'
         }).then((runs)=> {
-            if (!runs.length) {
+            if (!runs.length || runs[0].trashed) {
```

Comment:
The description of this strategy seems to indicate that it should actually return `runs[1]` if `runs[0]` is trashed but `runs[1]` matches the criteria.

Reply:
makes sense. I think in fact it shouldn't get runs[1] but whatever the most recent untrashed run is.